nss/integration-tests/integration_test.go

```diff
@@ -166,11 +166,13 @@ func TestMain(m *testing.M) {
 			// Purge .profraw files
 			files, _ := filepath.Glob(filepath.Join(currDir, "*.profraw"))
 			for _, f := range files {
-				os.Remove(f)
+				_ = os.Remove(f)
 			}
 		}()
 	}
 
-	m.Run()
+	code := m.Run()
 	testutils.MergeCoverages()
+
+	os.Exit(code)
```

Comment:
Imagine testutils.MergeCoverages() skip the test (for whatever reason), it’s better to defer os.Exit() just after m.Run() IMHO.

Reply:
There are some problems in the way we implemented the MergeCoverages though... There are some cases where we call `log.Fatalf()` and, from what I've seen, these calls completely ignore the defers that were queued in `TestMain`, so we should probably change that as well later (or it could be on this PR as well), wdyt?

Reply:
We introduce a copy of this code here, so I think it’s part of this PR to fix it.
You are correct that if we call log.Fatalf(), we have an issue (due to os.Exit() being called). Maybe we should return an error code then? The end result would be the same, but we at least know that this can stop early.

Reply:
Yeah, I think it's better. I'll update it.